sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryUtils.java

```diff
@@ -140,16 +140,23 @@ static void addAttribute(Span span, String key, Object value, OpenTelemetrySchem
      * @return the corresponding OpenTelemetry {@link Span}.
      */
     static Span setError(Span span, String statusMessage, Throwable throwable) {
-        if (throwable != null) {
-            span.recordException(throwable);
-            return span.setStatus(StatusCode.ERROR, statusMessage);
+        // "success" is needed for back compat with older Event Hubs and Service Bus, don't use it.
+        if ("success".equals(statusMessage)) {
+            statusMessage = null;
         }
 
-        // "success" is needed for back compat with older Event Hubs and Service Bus, don't use it.
-        if (statusMessage == null || "success".equals(statusMessage)) {
+        if (statusMessage == null && throwable == null) {
             return span;
         }
 
+        span.setAttribute(ERROR_TYPE_ATTRIBUTE,
+            statusMessage != null ? statusMessage : throwable.getClass().getName());
```

Comment:
just checking that `statusMessage` is low-cardinality?

also, if you want to copy the instrumentation-api logic, we do a bit of unwrapping of the throwable to get a slightly nicer / more stable name: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/DefaultErrorCauseExtractor.java

Reply:
yes, `statusMessage`  has low-cardinality - I added a comment to javadoc.
There is nothing that enforces it for traces though.

It's almost impossible to misuse it for metrics since we promote creating attributes and caching them in advance (vs creating on measurement).